mlir/include/mlir/Interfaces/LoopLikeInterface.td

```diff
@@ -235,6 +231,39 @@ def LoopLikeOpInterface : OpInterface<"LoopLikeOpInterface"> {
   }];
 
   let extraSharedClassDeclaration = [{
+    /// If there is a single induction variable return it, otherwise return
+    /// std::nullopt.
+    ::std::optional<::mlir::Value> getSingleInductionVar() {
```

Comment:
We should just deprecate these methods. THey never made sense to me.

Reply:
i agree. but i think that would require a whole lot of changes that i'm not keen on doing right now